src/services/loginService.ts

```diff
@@ -32,9 +33,20 @@ export class AzureLoginService {
     }
   }
 
-  public static async interactiveLogin(options: InteractiveLoginOptions): Promise<AuthResponse> {
-    await open("https://microsoft.com/devicelogin");
-    return await interactiveLoginWithAuthResponse(options);
+  public static async interactiveLogin(options?: InteractiveLoginOptions): Promise<AuthResponse> {
+    let authResp: AuthResponse = {credentials: undefined, subscriptions: []};
+    const fileTokenCache = new SimpleFileTokenCache();
+    if(!fileTokenCache.isEmpty()){
```

Comment:
Consider inverting the if/else to make the positive case on top.  If (isEmpty) // do thing 1 // otherwise do thing 2.